### PR TITLE
Submdspan bring in line with latest C++ draft (hidden friends)

### PR DESCRIPTION
**Pull request by crtrott**
- Make submdspan_mapping hidden friends.
- Take them out of the Proposed namespace since it is merged into draft.

### CODE REVIEW

**Review by mhoemmen — APPROVED** [review]
@crtrott and I discussed these changes offline.  They bring the implementation in line with the spec.

File: include/experimental/__p2630_bits/submdspan_mapping.hpp [hunk 1]
```diff
@@ -22,20 +22,18 @@
...
 
 namespace MDSPAN_IMPL_STANDARD_NAMESPACE {
-namespace MDSPAN_IMPL_PROPOSED_NAMESPACE {
 //******************************************
 // Return type of submdspan_mapping overloads
 //******************************************
-template <class Mapping> struct mapping_offset {
-  Mapping mapping;
+template <class LayoutMapping> struct submdspan_mapping_result {
+  _MDSPAN_NO_UNIQUE_ADDRESS LayoutMapping mapping = LayoutMapping();
```
Comment (dalg24) [inline include/experimental/__p2630_bits/submdspan_mapping.hpp]:
```suggestion
  _MDSPAN_NO_UNIQUE_ADDRESS LayoutMapping mapping{};
```

File: include/experimental/__p0009_bits/layout_left.hpp [hunk 2]
```diff
@@ -215,6 +215,16 @@ class layout_left::mapping {
 private:
    _MDSPAN_NO_UNIQUE_ADDRESS extents_type __extents{};
 
+   // [mdspan.submdspan.mapping], submdspan mapping specialization
+   template<class... SliceSpecifiers>
+     constexpr auto submdspan_mapping_impl(                    // exposition only
```
Comment (dalg24) [inline include/experimental/__p0009_bits/layout_left.hpp]:
```suggestion
     constexpr auto submdspan_mapping_impl(
```

File: include/experimental/__p0009_bits/layout_right.hpp [hunk 3]
```diff
@@ -217,6 +217,16 @@ class layout_right::mapping {
 private:
    _MDSPAN_NO_UNIQUE_ADDRESS extents_type __extents{};
 
+   // [mdspan.submdspan.mapping], submdspan mapping specialization
+   template<class... SliceSpecifiers>
+     constexpr auto submdspan_mapping_impl(                    // exposition only
```
Comment (dalg24) [inline include/experimental/__p0009_bits/layout_right.hpp]:
```suggestion
     constexpr auto submdspan_mapping_impl(
```

File: include/experimental/__p0009_bits/layout_stride.hpp [hunk 4]
```diff
@@ -489,6 +489,16 @@ struct layout_stride {
     }
 #endif
 
+   // [mdspan.submdspan.mapping], submdspan mapping specialization
+   template<class... SliceSpecifiers>
+     constexpr auto submdspan_mapping_impl(                    // exposition only
```
Comment (dalg24) [inline include/experimental/__p0009_bits/layout_stride.hpp]:
```suggestion
     constexpr auto submdspan_mapping_impl(
```

**Review by dalg24 — APPROVED** [review]

**Pull request merged**